examples/turbulent_advection-diffusion.jl

```diff
@@ -0,0 +1,181 @@
+using PassiveTracerFlows, Printf, Plots, JLD2, LinearAlgebra
+using Random: seed!
+
+# ## Choosing a device: CPU or GPU
+dev = CPU()
+nothing # hide
+
+# ## Setting up a `MultiLayerQG.Problem` to generate a turbulent flow
+#
+# The tubulent flow we use to advect the passive tracer is generated using the 
+# [`MultiLayerQG`](https://fourierflows.github.io/GeophysicalFlowsDocumentation/stable/modules/multilayerqg/) module
+# from the [`GeophysicalFlows.jl`](https://fourierflows.github.io/GeophysicalFlowsDocumentation/stable/) package.
+# A more detailed setup of this two layer system is found at the
+# [GeophysicalFlows Documentation](https://fourierflows.github.io/GeophysicalFlowsDocumentation/stable/literated/multilayerqg_2layer/).
+#
+# ### Numerical and time stepping parameters for the flow
+
+      n = 128            # 2D resolution = n²
+stepper = "FilteredRK4"  # timestepper
+     dt = 2.5e-3         # timestep
+
+# Physical parameters 
+L = 2π                   # domain size
+μ = 5e-2                 # bottom drag
+β = 5                    # the y-gradient of planetary PV
+
+nlayers = 2              # number of layers
+f₀, g = 1, 1             # Coriolis parameter and gravitational constant
+ H = [0.2, 0.8]          # the rest depths of each layer
+ ρ = [4.0, 5.0]          # the density of each layer
+
+ U = zeros(nlayers) # the imposed mean zonal flow in each layer
+ U[1] = 1.0
+ U[2] = 0.0
+
+# ### `MultiLayerQG.Problem` setup, shortcuts and initial conditions
+MQGprob = MultiLayerQG.Problem(nlayers, dev;
+                               nx=n, Lx=L, f₀, g, H, ρ, U, μ, β,
+                               dt, stepper, aliased_fraction=0)
+grid = MQGprob.grid
+x, y = grid.x, grid.y
+
+# Initial conditions                        
+seed!(1234) # reset of the random number generator for reproducibility
+q₀  = 1e-2 * ArrayType(dev)(randn((grid.nx, grid.ny, nlayers)))
+q₀h = MQGprob.timestepper.filter .* rfft(q₀, (1, 2)) # apply rfft  only in dims=1, 2
+q₀  = irfft(q₀h, grid.nx, (1, 2))                    # apply irfft only in dims=1, 2
+
+MultiLayerQG.set_q!(MQGprob, q₀)
+nothing 
+
+# ## Tracer advection-diffusion setup
+#
+# Now that we have a `MultiLayerQG.Problem` setup to generate our turbulent flow, we
+# setup an advection-diffusion simulation. This is done by passing the `MultiLayerQG.Problem`
+# as an argument to `TracerAdvectionDiffusion.Problem` which sets up an advection-diffusion problem
+# with same parameters where applicable. We also need to pass a value for the constant diffusivity `κ`,
+# the `stepper` used to step the problem forward and when we want the tracer released into the flow.
+# We will let the flow run until it reaches a statistical equilibrium and then advect-diffuse the tracer.
+
+κ = 0.002
+nsteps = 4000                    # total number of time-steps
+nsubs = 1                        # number of steps the simulation takes at each iteration 
```

Comment:
is this needed?

Reply:
No we can do without `nsubs`.

Reply:
Yes. Unless we use `nsubs` further down. As the example is now we don't.